residue_genomic_position_script.py

```diff
@@ -0,0 +1,266 @@
+import argparse
+from email import header
+from json import JSONDecodeError
+import logging
+import psutil
+
+import pandas as pd
+import requests
+from pandarallel import pandarallel
+import pyspark.sql.functions as f
+from pyspark.sql import SparkSession
+from pyspark.sql.types import StructField, StructType, IntegerType, StringType
+
+
+# Global configuration for Spark and Pandarallel.
+spark = SparkSession.builder.master('local[*]').getOrCreate()
+
+pandarallel.initialize(
+    nb_workers=psutil.cpu_count(),
+    progress_bar=True,
+)
+
+def main():
+
+    # PLIP INPUT (contain wanted data)
+    plip_json_input = (
+        # "gene_mapped_structures.json"
+        spark.read.json(args.plip_input)
+
+        .select("pdbStructureId", "chains", f.explode("compoundIds").alias("pdbCompoundId"))
+
+        .select("pdbStructureId", "pdbCompoundId", f.explode("chains").alias("chains"))
+
+        )
+
+    plip_json_input_v2 = (plip_json_input
+
+                        .withColumn("chainId", plip_json_input["chains.chainId"])
+
+                        .withColumn("geneId", plip_json_input["chains.geneId"])
+
+                        .withColumn("uniprotId", plip_json_input["chains.uniprot"])
+
+                        .drop("chains")
+
+                        )
+
+    # PLIP OUTPUT
+    plip_csv_output = (
+
+        # "output.csv"
+        spark.read.csv(args.plip_output, header=True, sep=",")
+
+        .withColumnRenamed("pdb_structure_id", "pdbStructureId")
+
+        .withColumnRenamed("compound_id", "pdbCompoundId")
+
+        .withColumnRenamed("prot_chain_id", "chainId")
+
```

Comment:
We need to add `.distinct()` here, because there are duplications in this data. We need to look into how it is generated to find out the reason.